int32 publish_count
int64[8] distances
